cmd/kured/main.go

```diff
@@ -344,7 +344,6 @@ func rebootAsRequired(nodeID string, window *timewindow.TimeWindow, TTL time.Dur
 			if err != nil {
 				log.Fatal(err)
 			}
-			nodeMeta.Unschedulable = node.Spec.Unschedulable
```

Comment:
This is not an unnecessary assignment.  The reason for this is to detect if the node was already cordoned such that when pod restarts it will remain cordoned.  (Don't uncordon the node just because of the reboot).

Now, if that is no longer intended, you would want to also change the code above such that it always uncordons the node and does not look at this metadata to make that choice (since, well, that metadata is clearly never set).

Reply:
Do we really want to touch the nodes if they have been cordonned? I would say no.

This is indeed a good point.

Reply:
This is a different question.

There are two things:

1)  If the node is already cordoned and we *are* going to reboot it, we should leave it cordoned after reboot
2)  If the node is already cordoned, do we just ignore it

Before this PR, the behavior was the same as (1) above except is had the problem of not draining the node before rebooting.  The point of this change was to not skip the drain if we are going to reboot, which is the simple removal of the conditional.  We still need to collect the cordoned state of the node such that it remains so after reboot.

However, if we want to make a new policy (I claim a breaking change) that cordoned nodes are not rebooted, then the logic needs to be very different and the uncordon above should not be gated based on the metadata in the lock.

I claim that the core behavior of the code was correct and acting like (1) and that the main change was to also drain any node that was going to reboot.

Reply:
Michael is right we don't want to change this value, I've reverted it. It's actually passed by reference to the `acquire()` func, so we need to make sure that the `nodeMeta` instance has the correct `Unschedulable` property value.

Reply:
@Michael-Sinz That was indeed my understanding, sorry if I haven't been verbose.

@jackfrancis ok, thanks!